examples/options/include/traccc/options/clusterization.hpp

```diff
@@ -8,27 +8,31 @@
 #pragma once
 
 // Local include(s).
+#include "traccc/clusterization/clustering_config.hpp"
 #include "traccc/options/details/config_provider.hpp"
 #include "traccc/options/details/interface.hpp"
 
 namespace traccc::opts {
 
 /// Options for the cell clusterization algorithm(s)
 class clusterization : public interface,
-                       public config_provider<unsigned short> {
+                       public config_provider<clustering_config> {
 
     public:
     /// Constructor
     clusterization();
 
     /// Configuration conversion
-    operator unsigned short() const override;
+    operator clustering_config() const override;
 
     private:
     /// @name Options
     /// @{
     /// The number of cells to merge in a partition
-    unsigned short target_cells_per_partition = 1024;
+    unsigned int threads_per_partition;
+    unsigned int max_cells_per_thread;
+    unsigned int target_cells_per_thread;
+    unsigned int backup_size_multiplier;
```

Comment:
I prefer "my" setup here, with giving default values to the properties in the header. And then using those in the `->default_value(...)` calls in the source file.